src/components/data/WdsTroubleshooter.js

```diff
@@ -0,0 +1,202 @@
+
+
+export const WdsTroubleshooter = ({ onDismiss, workspaceId, mrgId }) => {
+  const [leoOk, setLeoOk] = useState(null)
+  const [wdsResponsive, setWdsResponsive] = useState(null)
+  const [version, setVersion] = useState(null)
+  const [wdsStatus, setWdsStatus] = useState(null)
+  const [wdsDbStatus, setWdsDbStatus] = useState(null)
+  const [wdsPingStatus, setWdsPingStatus] = useState(null)
+  const [appFound, setAppFound] = useState(null)
+  const [appRunning, setAppRunning] = useState(null)
+  const [proxyUrl, setProxyUrl] = useState(null)
+  const [defaultInstanceExists, setDefaultInstanceExists] = useState(null)
+
+  const signal = useCancellation()
+
+  useOnMount(() => {
+    Ajax(signal).Apps.getV2AppInfo(workspaceId).then(res => {
+      setLeoOk(res)
+      const foundApp = resolveWdsApp(res)
+      setAppFound(foundApp?.appName)
+      setAppRunning(foundApp?.status)
+      setProxyUrl(foundApp?.proxyUrls?.wds)
+      Ajax(signal).WorkspaceData.getVersion(foundApp?.proxyUrls?.wds).then(res => {
+        setWdsResponsive(true)
+        setVersion(res.git?.commit?.id)
+      }).catch(_ => {
+        setWdsResponsive(false)
+        setVersion('unknown')
+      })
+      Ajax(signal).WorkspaceData.getStatus(foundApp?.proxyUrls?.wds).then(res => {
+        setWdsStatus(res.status)
+        setWdsDbStatus(res.components?.db?.status)
+        setWdsPingStatus(res.components?.ping?.status)
+      }).catch(_ => {
+        setWdsStatus('unresponsive')
+        setWdsDbStatus('unknown')
+        setWdsPingStatus('unknown')
+      })
+      Ajax(signal).WorkspaceData.listInstances(foundApp?.proxyUrls?.wds).then(res => {
+        setDefaultInstanceExists(res.includes(workspaceId))
+      }).catch(_ => {
+        setDefaultInstanceExists('unknown')
+      })
+    }).catch(_ => {
+      setLeoOk([])
+      setAppFound('unknown')
+      setAppRunning('unknown')
+      setProxyUrl('unknown')
+      setWdsResponsive('unknown')
+      setVersion('unknown')
+      setWdsStatus('unresponsive')
+      setWdsDbStatus('unknown')
+      setWdsPingStatus('unknown')
+    })
+  })
+
+  const checkIcon = (status, size = 24) => Utils.switchCase(status,
+    ['success', () => icon('success-standard', { size, style: { color: colors.success() }, 'aria-label': 'Validation Success' })],
+    ['failure', () => icon('error-standard', { size, style: { color: colors.danger(0.85) }, 'aria-label': 'Validation Failure' })],
+    ['running', () => icon('loadingSpinner', { size, style: { color: colors.primary() }, 'aria-label': 'Validation Running' })]
+  )
+
+  const clippy = (label, value) => !!value && h(Fragment, [value, h(ClipboardButton, {
+    'aria-label': `Copy ${label} to clipboard`,
+    style: { marginLeft: '1rem' },
+    text: value
+  })])
+
+  const troubleShooterText = {
+    'Workspace Id':	workspaceId,
+    'Resource Group Id':	mrgId,
+    'App listing':	leoOk?.length,
+    'Data Table app found':	appFound,
+    'Data Table app running?':	appRunning,
```

Comment:
Also changed all the references to 'WDS' to 'Data Table' but it does seem a bit awkward, so happy for suggestions for better wording

Reply:
I think it's ok. How about "Data app xxxx" or "Data service xxxx"?